src/rust/receiver-mock/src/router.rs

```diff
@@ -1,216 +1,283 @@
-                    string_body = String::from_utf8(vector_body).unwrap();
-                }
 
-                let lines = string_body.trim().lines();
-
-                let content_type = headers
-                    .get("content-type")
-                    .unwrap_or(&empty_header)
-                    .to_str()
-                    .unwrap();
-                match content_type {
-                    // Metrics in carbon2 format
-                    "application/vnd.sumologic.carbon2" => {
-                        metrics::handle_carbon2(lines, address, &stats, opts.print);
-                    }
-                    // Metrics in graphite format
-                    "application/vnd.sumologic.graphite" => {
-                        metrics::handle_graphite(lines, address, &stats, opts.print);
-                    }
-                    // Metrics in prometheus format
-                    "application/vnd.sumologic.prometheus" => {
-                        metrics::handle_prometheus(lines, address, &stats, opts.print);
-                    }
-                    // Logs & events
-                    "application/x-www-form-urlencoded" => {
-                        if opts.print.logs {
-                            let mut counter = 0;
-                            for line in lines {
-                                println!("log => {}", line);
-                                counter += 1;
-                            }
-                            let mut stats = stats.lock().unwrap();
-                            (*stats).logs += counter;
-                            (*stats).logs_bytes += vector_length;
-
-                            let logs_ip_list =
-                                (*stats).logs_ip_list.entry(address).or_insert((0, 0));
-                            (*logs_ip_list).0 += counter;
-                            (*logs_ip_list).1 += vector_length;
-                        } else {
-                            let lines_count = lines.count() as u64;
-                            let mut stats = stats.lock().unwrap();
-                            (*stats).logs_bytes += vector_length;
-                            (*stats).logs += lines_count;
-
-                            let logs_ip_list =
-                                (*stats).logs_ip_list.entry(address).or_insert((0, 0));
-                            (*logs_ip_list).0 += lines_count;
-                            (*logs_ip_list).1 += vector_length;
-                        }
-                    }
-                    &_ => {
-                        println!("invalid header value");
-                    }
+        // Metrics in prometheus format
+        "application/vnd.sumologic.prometheus" => {
+            let result = metrics::handle_prometheus(lines, remote_address, opts.print);
+            app_state.add_metrics_result(result);
+        }
+
+        // Logs & events
+        "application/x-www-form-urlencoded" => {
+            // TODO: refactor
```

Comment:
Strange ToDo taking into account the PR name 😹

Reply:
🤷🏻  I don't really want to touch too much stuff in this PR (it's already relatively big)

Reply:
What I meant here is that looking at how metrics are handled above this can be done in a similar fashion for logs